form/src/main/java/org/dhis2/form/ui/binding/OptionSetSelectionBindings.kt

```diff
@@ -0,0 +1,142 @@
+            visibility = View.VISIBLE
+        }
+        else -> visibility = View.GONE
+    }
+}
+
+@BindingAdapter("renderingType")
+fun RadioGroup.setRenderingType(renderingType: UiRenderType?) {
+    when (renderingType) {
+        UiRenderType.VERTICAL_RADIOBUTTONS -> {
+            orientation = LinearLayout.VERTICAL
+            visibility = View.VISIBLE
+        }
+        UiRenderType.HORIZONTAL_RADIOBUTTONS -> {
+            orientation = LinearLayout.HORIZONTAL
+            visibility = View.VISIBLE
+        }
+        else -> visibility = View.GONE
+    }
+}
+
+@BindingAdapter("options")
+fun LinearLayout.addOptions(item: FieldUiModel) {
+    if (item.renderingType?.isCheckBox() == true) {
+        removeAllViews()
+        item.options?.filter { item.canShowOption(it.uid()) }?.forEach { option ->
+            val optionBinding: OptionSetSelectCheckItemBinding =
+                OptionSetSelectCheckItemBinding.inflate(LayoutInflater.from(context), this, false)
+            optionBinding.apply {
+                this.item = item
+                this.option = option
+                checkBox.apply {
+                    val optionChecked = item.displayName == option.displayName()
+                    isChecked = optionChecked
+                    setOnCheckedChangeListener { _, isChecked ->
+                        when {
+                            isChecked -> item.onSave(option.code())
+                            else -> item.takeIf { optionChecked }?.onClear()
+                        }
+                    }
+                }
+            }
+            addView(optionBinding.root)
+        }
+    }
+}
+
+@BindingAdapter("options")
+fun RadioGroup.addOptions(item: FieldUiModel) {
+    if (item.renderingType?.isRadioButton() == true) {
+        removeAllViews()
+        item.options?.filter { item.canShowOption(it.uid()) }?.forEach { option ->
+            val optionBinding: OptionSetSelectItemBinding =
+                OptionSetSelectItemBinding.inflate(LayoutInflater.from(context), this, false)
+            optionBinding.apply {
+                this.item = item
+                this.option = option
+                radio.apply {
+                    isChecked = item.displayName == option.displayName()
+                    setOnCheckedChangeListener { _, isChecked ->
+                        when {
+                            isChecked -> item.onSave(option.code())
+                        }
+                    }
+                }
+            }
+            addView(optionBinding.root)
+        }
+    }
+}
+
+private fun FieldUiModel.canShowOption(optionUid: String): Boolean {
+    val inOptionsToShow: Boolean = optionsToShow?.contains(optionUid) ?: false
+    val inOptionsToHide: Boolean = optionsToHide?.contains(optionUid) ?: false
+    return when {
+        optionsToShow?.isNotEmpty() ?: false -> inOptionsToShow
+        else -> !inOptionsToHide
+    }
+}
+
+@BindingAdapter("optionTint")
```

Comment:
Is this specific from Optionset or can be generify to other components?
See `FieldBindigns.kt ->@BindingAdapter("input_layout_style")`

Reply:
Yes, maybe it could be used by other components, I will move it